build/logrotator.go

```diff
@@ -73,6 +74,13 @@ func (r *RotatingLogWriter) InitLogRotator(logFile string, maxLogFileSize int,
 		return fmt.Errorf("failed to create file rotator: %w", err)
 	}
 
+	c, err := zstd.NewWriter(nil)
+	if err != nil {
+		return fmt.Errorf("failed to create zstd compressor: %w", err)
+	}
+
+	r.logRotator.SetCompressor(c, "zst")
```

Comment:
As mentioned offline: How widely available are tools like `zgrep`, `zcat` and `zless` that support this format in the main Unix variants?
Maybe it would make sense to allow users to configure this.

Reply:
zutils (which provides `zgrep` and `zcat`) added `zstd` support in version 1.11 in January 2022:

https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=1009305

Which is in Debian Bookworm (current stable), as well as the backports repos for oldstable.

The `zstd` package is in the oldoldstable, oldstable, and stable repos. That provides equivalent commands like `zstdgrep`, `zstdcat`, and `zstdless`.

On my system (debian testing), `zless` is a shell script wrapper around `less` that only supports `gzip`. It seems like that command is compressor-specific, as there is also `xzless` (and `zstdless`).

Reply:
Darwin / macOS 14.6 zgrep, zless support .zst file inputs
zcat is unsupported

Reply:
Worth noting that on Debian I had to install an explicit package `zutils` to get newer versions of these programs - not sure what the source package was for the older versions.